CMakeLists.txt

```diff
@@ -133,6 +133,26 @@ configure_file("source/Version.h.in" "${PROJECT_BINARY_DIR}/Version.h")
 # OpenSSL dependency                    #
 #########################################
 set(OPENSSL_USE_STATIC_LIBS TRUE)
+
+if(APPLE)
```

Comment:
At this moment, Device Client does not provide support for MacOS. This is mainly because of the shadow feature of device client which does not work on MacOS because of dependency issue which is expected and known. Even if your change over here solves the issue of finding the OpenSSL on MacOS, we cannot push this change since it will give the wrong impression of device client is being supported on MacOS which is not true. For this reason we will have to close this review. 

When we add support for Device Client on MacOS, we can reopen this review and merge your changes. 

Thank you.